### ISSUE TITLE
Renew LetsEncrypt Certificate

### ISSUE THREAD

**Issue by AlphaGeeKUK** [Microsoft/nav-docker#217]
How can I easily renew a LetsEncrypt certificate in a container deployed with aka.ms/getnav using NAV 2018 GB CU6 image?

**Comment by freddydk** [comment]
I didn't create a Cmdlet to do that, after all, the expectation is that the VM is renewed within 90 days anyway. That said, I will probably create the function, I just haven't had a machine on my hand with an expired certificate yet:-)

**Comment by AlphaGeeKUK** [comment]
Lucky you :-)

The machine I will need to renew it on is going to be needed for longer than 90 days so a Cmdlet would be awesome.

**Comment by freddydk** [comment]
Lucky you :-)

I will look into this within the next 90 days:-)

**Comment by AlphaGeeKUK** [comment]
Ha ha. Thanks Freddy 👍

**Comment by Koubek** [comment]
Actually, you can renew whenever you want Freddy. Even in the case, when your certificate has been issued recently.

The only issue can be a rate limit. For this reason, I would strongly recommend watching and using this during the development (I have a config in my Let's Encrypt PowerShell scripts and I just switch over PRO to staging and vice-versa).

**Comment by freddydk** [comment]
In the next version of the navcontainerhelper, you will be able to run this script:

```
. "C:\myfolder\SetupCertificate.ps1"
Renew-LetsEncryptCertificate -publicDnsName $dnsidentity -CertPfxFilename $certificatePfxFile -CertPfxPassword $certificatePfxPassword
Restart-NavContainer -containerName navserver -renewBindings
```

The SetupCertificate.ps1 contains the info needed for a renewal.
This will renew the lets encrypt certificate for a fresh 90 day period.

**Comment by freddydk** [comment]
This is fixed in navcontainerhelper 0.0.3.0, which is latest now.
